### ISSUE TITLE
Assign card list missing search bar with 8+ cards

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#51906]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** Needs reproduction
**Reproducible in production?:** Needs reproduction
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @joekaufmanexpensify 
**Slack conversation** (hyperlinked to channel name):  [quality](https://applause-ts.slack.com/archives/C05LX9D6E07/p1730402792152209)

## Action Performed:
Prerequisite:
Members and company cards added in workspace , 8+cards added to the account

1. Go to staging.new.expensify.com
2. Choose the workspace from settings
3. Start assign card flow

## Expected Result:
Should have search field if there are 8+ cards
## Actual Result:
Search field is missing
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

https://github.com/user-attachments/assets/74abfd81-f34c-44e0-b413-17a8f18bb2fd

  
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by m-natarajan** [comment]
BZ team member, Callstack will work on this issue, please wait for them to comment before moving the issue forward.

**Comment by mountiny** [comment]
Handling this as part of the project I have worked on

**Comment by VickyStash** [comment]
Hi, I'm Viktoryia from Callstack - expert contributor group - and I would like to work on this issue.

**Comment by joekaufmanexpensify** [comment]
assigned!

**Comment by VickyStash** [comment]
I've prepared a [Draft PR](https://github.com/Expensify/App/pull/51955), but there are some problems with open keyboard views on mobile, especially on small devices. I'm going to improve it tomorrow.

**Comment by VickyStash** [comment]
It looks like to fix the keyboard overlapping issue we need to change the scrolling on the cards selection screen.

Right now, the description text and the search bar are fixed, which makes it be overlapped by the keyboard:

https://github.com/user-attachments/assets/1c1e7ca8-b238-486c-b694-0647e56e8d11

But if we make the screen scrollable overall, we at least can see the input:

https://github.com/user-attachments/assets/e04bc3d9-b4c0-490c-8192-300325939568

@joekaufmanexpensify @mountiny Please, let me know what do you think and if you have any other ideas about the way it should look, thank you!

**Comment by mountiny** [comment]
cc @Expensify/design which appraoch do we use across the app here

**Comment by dubielzyk-expensify** [comment]
Hmm. I don't think we should even fix the (1)-(2)-(3)-(4) which would make everything scrollable except for the top header with the back button, but keen to hear what the other designers think.

**Comment by shawnborton** [comment]
Oh boy this is a tricky one. Do we have other examples of how this is currently working in the app, where we have a progress stepper + search bar + list? 

I don't feel too strongly whether we keep the numbers at the top fixed or not - happy to let @dubielzyk-expensify and @dannymcclain decide that one.

**Comment by VickyStash** [comment]
So @dubielzyk-expensify @dannymcclain should I try to make the page fully scrollable for this step? Including stepper?

**Comment by dannymcclain** [comment]
Yes I agree with Jon. I think _only_ the header should be fixed and we can let the rest of the page scroll.
![CleanShot 2024-11-06 at 07 58 42@2x](https://github.com/user-attachments/assets/358c0be6-b613-41dd-8e47-534d4aedea23)

**Comment by VickyStash** [comment]
The [PR](https://github.com/Expensify/App/pull/51955) is ready for the review!

**Comment by mountiny** [comment]
PR merged

**Comment by mountiny** [comment]
@dukenv0307 can you please add the checklist but I dont think we need a test for this one specific bug, then we can close as payments will be handled separately
